tools/src/main/python/dlpx/virtualization/_internal/delphix_client.py

```diff
@@ -20,10 +22,14 @@ class DelphixClient(object):
     """
     __BOUNDARY = '----------boundary------'
     __UPLOAD_CONTENT = 'multipart/form-data; boundary={}'.format(__BOUNDARY)
+    __WAIT_TIME_SECONDS = 5
+    __WAIT_TIMEOUT_SECONDS = 1800
```

Comment:
was there any discussion on the number of seconds here? is 30 mins long enough? I think when we were doing some calculations we did say 30 mins was the upper-bound but wanted to make sure that we thought about this.

another option, which might be worth considering for a different bug would be to allow the cli user to pass in a number to represent the number of seconds to wait. so something like:
`--wait 1800`?

Reply:
Yeah, we decided to go with 30 minutes because that was the upper bound we saw from performance testing a while back. I considered having the end user pass in a value for how long to wait, but thought that it was kind of weird since it's hard to predict how long the upgrade takes from the end-user's perspective? I don't mind changing the code to accommodate for custom timeouts, but thought that what I have right now was enough for our purposes.

Reply:
cool I was thinking thinking more the case where if a user was like. i have a max of 1 hour before I need to do operations on the dsource or vdb, so the upgrade could take up to that long, it would really suck if say their upgrade would have took 31 mins and then it got canceled and then say they tried to run it but now they have to cancel it manually if the full hour goes by? (so more like.. if things are scheduled idk)

although just to confirm, if it does actually timeout, do we cancel the upgrade or do we just not monitor the result anymore?